### ISSUE TITLE
sofmax layer

### ISSUE THREAD

**Issue by itachi176** [Tencent/ncnn#4683]
## detail | 详细描述 | 詳細な説明
I have a model segmentation. output of model is tensor (1,4,768,768) need foward to sofmax layer before argmax. this is my code:

   ncnn::Layer* softmax;
    softmax = ncnn::create_layer(ncnn::LayerType::Softmax);
    ncnn::ParamDict pd;
    ncnn::Option options;
    pd.set(0, 1);
    softmax->load_param(pd);
    softmax->forward(out, out, options);

I try to use sofmax at dim = 1 but it return "param is too old, please regenerate!". it doesn't give an error but i think the result will be incorrect. is there any idea to fix this or some trick to implement sofmax when inference segment model like this?

**Comment by nihui** [comment]
`
pd.set(1, 1);
`

btw, it is suggested to add softmax operation into your param file directly. It will be accelerated with optimal memory layout/ data type and vulkan automatically.

**Comment by itachi176** [comment]
> 

How can I add Softmax operation to my param file? I think to add softmax to the model when training in Pytorch. Have the other way?

**Comment by itachi176** [comment]
> `pd.set(1, 1);`
> 
> btw, it is suggested to add softmax operation into your param file directly. It will be accelerated with optimal memory layout/ data type and vulkan automatically AND it is much easier than calling low-level op api
How can I add Softmax operation to my param file? I think to add softmax to the model when training in Pytorch. Have the other way?

**Comment by itachi176** [comment]
> `pd.set(1, 1);`
> 
> btw, it is suggested to add softmax operation into your param file directly. It will be accelerated with optimal memory layout/ data type and vulkan automatically AND it is much easier than calling low-level op api

How can I add Softmax operation to my param file? I think to add softmax to the model when training in Pytorch. Have the other way?

**Comment by nihui** [comment]
open param with your text editor, add line

```
Softmax                  softmax                     1 1 in out 0=1 1=1
```

and then adjust the leading number of layer count and blob count

https://github.com/Tencent/ncnn/wiki/param-and-model-file-structure

**Comment by nihui** [comment]
https://github.com/Tencent/ncnn/wiki/FAQ-ncnn-throw-error#param-is-too-old-please-regenerate
